### PR TITLE
chore(lib): Add library template

### PR DESCRIPTION
**Pull request by damassi**
This sets up the library example repo with some standard Artsy defaults, along with the necessary plumbing to connect to our NPM publishing pipeline. 

Includes: 

- TypeScript
- ESLint 
- Jest 
- Prettier 

The README covers the basics of our pipeline; let me know if I should add more detail.

### CODE REVIEW

File: .autorc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "extends": "@artsy",
+  "plugins": [
+    [
+      "npm",
+      {
+        "publishFolder": "./dist"
```
Comment (dzucconi) [inline .autorc]: I think you can remove this?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+```
+
+## Publishing Overview
+
+There are a few key tools that we use for our publishing pipeline. Notably
+
+- [Auto](https://github.com/intuit/auto): Generate releases based on semantic version labels on pull requests
+- [Artsy's CircleCI Orbs](https://github.com/artsy/orbs): Orbs are packages of CircleCI configuration that can be shared across projects
+
+Combining these things together, we get seamless DX around releases that can be triggered directly from PRs via labels, and minimal CI config for running common script commands in an ideal way. Additionally, an automated changelog is generated for every release.
```
Comment (dzucconi) [inline README.md:22]: We should have a step by step for actually setting up a project.

For instance: you have to set up the project in Circle

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export const computeTwoNumbers = (a: number, b: number): number => {
+  return a + b
+}
```
Comment (dzucconi) [inline src/index.ts:3]: 👍

File: package.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "@artsy/library-example",
+  "private": true,
+  "version": "0.0.2",
```
Comment (dzucconi) [inline package.json]: Version

File: package.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "@artsy/library-example",
+  "private": true,
```
Comment (dzucconi) [inline package.json]: We should maybe make sure this actually works?

File: README.md (hunk 2 above)
Comment (damassi) [reply README.md:22]: Ah, right. Anything else you can remember from recent memory that you had to do in icons? Or is Circle addition it?

File: package.json (hunk 5 above)
Comment (damassi) [reply package.json]: Need to remove this

File: README.md (hunk 2 above)
Comment (dzucconi) [reply README.md:22]: Oh, give Artsy repo permissions

Comment (damassi) [reply README.md:22]: Updated

**Pull request merged**